File: LeetCode/evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        r=[]
        for i in tokens:
            if i not in '+*/-':
                r.append(int(i))
            else:
                b=r.pop()
                a=r.pop()
                if i=='+':
                    r.append(a+b)
                elif i=='-':
                    r.append(a-b)
                elif i=='*':
                    r.append(a*b)
                elif i=='/':
                    r.append(int(a/b))
        return r.pop()

File: LeetCode/evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> op =new Stack<>();  
        for(String s:tokens){
            if("+/-*".contains(s)){
                int b=op.pop();
                int a=op.pop();
                if(s.equals("+")) op.push(a+b);
                else if(s.equals("-")) op.push(a-b);
                else if(s.equals("/")) op.push(a/b);
                else if(s.equals("*")) op.push(a*b);
            }
            else{
                op.push(Integer.parseInt(s));
            }
        } 
        return(op.peek());
    }
}